include/aligator/modelling/dynamics/centroidal-fwd.hpp

```diff
@@ -0,0 +1,67 @@
+/// @copyright Copyright (C) 2022 LAAS-CNRS, INRIA
+#pragma once
+
+#include "aligator/modelling/dynamics/ode-abstract.hpp"
+
+#include <proxsuite-nlp/modelling/spaces/vector-space.hpp>
+
+namespace aligator {
+namespace dynamics {
+template <typename Scalar> struct CentroidalFwdDataTpl;
+
+/**
+ * @brief   Nonlinear centroidal forward dynamics.
+ *
+ * @details This is described in state-space \f$\mathcal{X} = T\mathcal{Q}\f$
+ * (the phase space \f$x = (c,h,L)\f$ with c CoM position, h linear momentum)
+ * and L angular momentum) using the differential equation \f[ \dot{x}
+ * = f(x, u) = F_x * x + F_u(x) * u \f] as described by the Newton-Euler law
+ * of momentum.
+ *
+ */
+template <typename _Scalar>
+struct CentroidalFwdDynamicsTpl : ODEAbstractTpl<_Scalar> {
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  using Scalar = _Scalar;
+  ALIGATOR_DYNAMIC_TYPEDEFS(Scalar);
+  using Base = ODEAbstractTpl<Scalar>;
+  using BaseData = ODEDataTpl<Scalar>;
+  using ContDataAbstract = ContinuousDynamicsDataTpl<Scalar>;
+  using Data = CentroidalFwdDataTpl<Scalar>;
+  using Manifold = proxsuite::nlp::VectorSpaceTpl<Scalar>;
+  using ManifoldPtr = shared_ptr<Manifold>;
+  using Matrix3s = Eigen::Matrix<Scalar, 3, 3>;
+
+  using Base::nu_;
+
+  ManifoldPtr space_;
+  const std::size_t nk_;
+  const double mass_;
```

Comment:
Data members should not be const: 
```suggestion
  std::size_t nk_;
  double mass_;
```
Const data members [disable the copy-assignment operator](https://stackoverflow.com/questions/26131921/const-class-member-copy-constructor)